drivers/firmware/psci/psci.c

```diff
@@ -442,7 +442,8 @@ static void __init psci_0_2_set_functions(void)
 
 	arm_pm_restart = psci_sys_reset;
 
-	pm_power_off = psci_sys_poweroff;
+	if (!pm_power_off)
+		pm_power_off = psci_sys_poweroff;
```

Comment:
why not `psci_sys_reset`? it seems to lead to `REBOOT_WARM || REBOOT_SOFT`, aren't we interested in that instead of poweroff the platform?

Reply:
ok, now I understand (I beleve): this one is called by kernel/reboot machine_power_off() and firmware sets this unconditionaly overriding previous handler which you set in restart-poweroff driver unconditionally in this PR